allennlp/data/dataset.py

```diff
@@ -77,7 +77,8 @@ def get_padding_lengths(self) -> Dict[str, Dict[str, int]]:
 
     def as_arrays(self,
                   padding_lengths: Dict[str, Dict[str, int]] = None,
-                  verbose: bool = True) -> Dict[str, List[numpy.array]]:
+                  verbose: bool = True) -> Dict[str, List[Union[numpy.array,
```

Comment:
(1) yes, I think generics would work here in the same way, it shouldn't be too much work
(2) yes, as of my last commit, mypy works on the whole codebase, I'll add it to travis (I might need to add it to `requirements.txt` too